prog/dns_zone/dns_zone_nexus.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require "forwardable"
+
+require_relative "../../lib/util"
+
+class Prog::DnsZone::DnsZoneNexus < Prog::Base
+  subject_is :dns_zone
+
+  label def wait
+    dns_zone.dns_servers.each do |dns_server|
+      records_to_rectify = dns_zone.records_dataset
+        .left_join(:seen_dns_records_by_dns_servers, dns_record_id: :id, dns_server_id: dns_server.id)
+        .where(Sequel[:seen_dns_records_by_dns_servers][:dns_record_id] => nil).all
+
+      next if records_to_rectify.empty?
+
+      commands = ["zone-abort #{dns_zone.name}", "zone-begin #{dns_zone.name}"]
+      records_to_rectify.each do |r|
+        commands << if r.tombstoned
```

Comment:
after making sure all the tombstoned records are unset in the dns_zone, I think we should destroy the records. Keeping them as is in the DB with a tombstoned column is also OK but different than how we manage the entity lifecycle at the moment.

Reply:
that makes sense, but I'll implement this in a next PR as there are other PRs depends on this.